lxd/storage_zfs.go

```diff
@@ -1842,75 +1842,73 @@ func (s *storageZfs) zfsPoolCreate() error {
 
 	// Create default dummy datasets to avoid zfs races during container
 	// creation.
-	err := s.zfsPoolVolumeCreate("containers")
-	if err != nil {
-		return err
-	}
-
-	err = s.zfsPoolVolumeSet("containers", "mountpoint", "none")
+	poolName := s.getOnDiskPoolName()
+	dataset := fmt.Sprintf("%s/containers", poolName)
+	fixperms := shared.VarPath("storage-pools", s.pool.Name, "containers")
+	msg, err := zfsPoolVolumeCreate(dataset, fmt.Sprintf("mountpoint=%s", fixperms))
```

Comment:
This is for the case where the user has set mountpoints on already existing dataset that would be inherited by the suddatasets. I do however, now that I see this, think that all of those subdatasets should have `mountpoint=none` including the pool when reusing an existing one.